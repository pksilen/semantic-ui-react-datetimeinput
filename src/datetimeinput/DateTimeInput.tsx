import * as React from 'react';
import { DateTimeInputProps, DateTimeInputPropTypes, DateTimeOptionalInputProps } from './DateTimeInputProps';
import TimeInput from '../timeinput/TimeInput';
import DateInput from '../dateinput/DateInput';
import { TimeInputProps } from '../timeinput/TimeInputProps';
import { DateInputProps } from '../dateinput/DateInputProps';

const DateTimeInput = ({
  onDateValueChange,
  dateValue,
  buttonPlacement,
  doubleClickDelayInMillis,
  focusColor,
  showTooltipDelayInMillis,
  showTooltips,
  size,
  valueType,
  hourSingleClickStepCount,
  hourDoubleClickStepCount,
  minuteSingleClickStepCount,
  minuteDoubleClickStepCount,
  daySingleClickStepCount,
  dayDoubleClickStepCount,
  monthSingleClickStepCount,
  monthDoubleClickStepCount,
  yearSingleClickStepCount,
  yearDoubleClickStepCount,
  dateFormat,
  ...restOfProps
}: DateTimeInputProps): React.ReactElement<DateTimeInputProps> => {
  const timeInputProps: TimeInputProps = {
    onDateValueChange,
    dateValue,
    buttonPlacement,
    doubleClickDelayInMillis,
    focusColor,
    showTooltipDelayInMillis,
    showTooltips,
    size,
    valueType,
    hourSingleClickStepCount,
    hourDoubleClickStepCount,
    minuteSingleClickStepCount,
    minuteDoubleClickStepCount
  };

  const dateInputProps: DateInputProps = {
    onDateValueChange,
    dateValue,
    buttonPlacement,
    doubleClickDelayInMillis,
    focusColor,
    showTooltipDelayInMillis,
    showTooltips,
    size,
    valueType,
    daySingleClickStepCount,
    dayDoubleClickStepCount,
    monthSingleClickStepCount,
    monthDoubleClickStepCount,
    yearSingleClickStepCount,
    yearDoubleClickStepCount,
    dateFormat
  };

  return (
    <div {...restOfProps}>
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <DateInput className="dateInput" {...dateInputProps} />
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <TimeInput className="timeInput" {...timeInputProps} />
    </div>
  );
};

const propTypes: DateTimeInputPropTypes = {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  ...TimeInput.propTypes,
  // eslint-disable-next-line react/forbid-foreign-prop-types
  ...DateInput.propTypes,
};

DateTimeInput.propTypes = propTypes;

const defaultProps: DateTimeOptionalInputProps = {
  ...TimeInput.defaultProps,
  ...DateInput.defaultProps
};

DateTimeInput.defaultProps = defaultProps;

export default DateTimeInput;
