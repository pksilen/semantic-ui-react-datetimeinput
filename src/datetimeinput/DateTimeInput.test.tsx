import * as React from 'react';
import PropTypes from 'prop-types';
import { mount, shallow as renderShallow } from 'enzyme';
import DateTimeInput from './DateTimeInput';

const date = new Date('2019-01-10T12:30:00.000');
let onDateValueChangeMock: (newValue: Date) => void;

beforeEach(() => {
  onDateValueChangeMock = jest.fn();
});

describe('propTypes', () => {
  it('should validate prop types', () => {
    // eslint-disable-next-line react/forbid-foreign-prop-types
    const { propTypes } = DateTimeInput;

    expect(propTypes.daySingleClickStepCount).toBe(PropTypes.number);
    expect(propTypes.dayDoubleClickStepCount).toBe(PropTypes.number);
    expect(propTypes.monthSingleClickStepCount).toBe(PropTypes.number);
    expect(propTypes.monthDoubleClickStepCount).toBe(PropTypes.number);
    expect(propTypes.yearSingleClickStepCount).toBe(PropTypes.number);
    expect(propTypes.yearDoubleClickStepCount).toBe(PropTypes.number);
    expect(propTypes.dateFormat).toBeTruthy();
    expect(propTypes.hourSingleClickStepCount).toBe(PropTypes.number);
    expect(propTypes.hourDoubleClickStepCount).toBe(PropTypes.number);
    expect(propTypes.minuteSingleClickStepCount).toBe(PropTypes.number);
    expect(propTypes.minuteDoubleClickStepCount).toBe(PropTypes.number);
  });
});

describe('defaultProps', () => {
  it('should set default prop values correctly', () => {
    const dateTimeInputWrapper = mount(
      <DateTimeInput dateValue={new Date()} onDateValueChange={onDateValueChangeMock} />
    );

    expect(dateTimeInputWrapper.props().daySingleClickStepCount).toBe(1);
    expect(dateTimeInputWrapper.props().dayDoubleClickStepCount).toBe(7);
    expect(dateTimeInputWrapper.props().monthSingleClickStepCount).toBe(1);
    expect(dateTimeInputWrapper.props().monthDoubleClickStepCount).toBe(3);
    expect(dateTimeInputWrapper.props().yearSingleClickStepCount).toBe(1);
    expect(dateTimeInputWrapper.props().yearDoubleClickStepCount).toBe(5);
    expect(dateTimeInputWrapper.props().dateFormat).toBe('D M YYYY');
    expect(dateTimeInputWrapper.props().hourSingleClickStepCount).toBe(1);
    expect(dateTimeInputWrapper.props().hourDoubleClickStepCount).toBe(4);
    expect(dateTimeInputWrapper.props().minuteSingleClickStepCount).toBe(1);
    expect(dateTimeInputWrapper.props().minuteDoubleClickStepCount).toBe(5);
  });
});

describe('render()', () => {
  it('should render component correctly', () => {
    const timeInputWrapper = renderShallow(
      <DateTimeInput dateValue={date} onDateValueChange={onDateValueChangeMock} />
    );

    expect(timeInputWrapper).toMatchSnapshot();
  });

  it('should render additional props to outside div', () => {
    const dateTimeInputWrapper = mount(
      <DateTimeInput
        id="testId"
        className="testClassName"
        dateValue={date}
        onDateValueChange={onDateValueChangeMock}
      />
    );

    const outsideDivWrapper = dateTimeInputWrapper.find('div').first();

    expect(outsideDivWrapper.props().id).toBe('testId');
    expect(outsideDivWrapper.props().className).toBe('testClassName');
  });
});
