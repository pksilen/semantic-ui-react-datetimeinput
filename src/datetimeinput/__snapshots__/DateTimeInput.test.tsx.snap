// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() should render component correctly 1`] = `
<div>
  <DateInput
    buttonPlacement="buttonsOutside"
    className="dateInput"
    dateFormat="D M YYYY"
    dateValue={2019-01-10T10:30:00.000Z}
    dayDoubleClickStepCount={7}
    daySingleClickStepCount={1}
    doubleClickDelayInMillis={250}
    focusColor="#85b7d9"
    monthDoubleClickStepCount={3}
    monthSingleClickStepCount={1}
    onDateValueChange={[MockFunction]}
    showTooltipDelayInMillis={500}
    showTooltips={true}
    size="small"
    valueType="start"
    yearDoubleClickStepCount={5}
    yearSingleClickStepCount={1}
  />
  <TimeInput
    buttonPlacement="buttonsOutside"
    className="timeInput"
    dateValue={2019-01-10T10:30:00.000Z}
    doubleClickDelayInMillis={250}
    focusColor="#85b7d9"
    hourDoubleClickStepCount={4}
    hourSingleClickStepCount={1}
    minuteDoubleClickStepCount={5}
    minuteSingleClickStepCount={1}
    onDateValueChange={[MockFunction]}
    showTooltipDelayInMillis={500}
    showTooltips={true}
    size="small"
    valueType="start"
  />
</div>
`;
